admin/class-acc_user_importer-admin.php

```diff
@@ -356,94 +301,49 @@ private function proccess_user_data ( $users ) {
-			$this->log_dual("Received " . $user_info);
+			$userInfoString = $userFirstName . " " . $userLastName;
+			$userInfoString .= " " . $userEmail;
+			$userInfoString .= " home:" . $userHomePhone;
+			$userInfoString .= " cell:" . $userCellPhone;
+			$userInfoString .= " member#:" . $userMembership;
+			$userInfoString .= " expiry:" . $userExpiry;
+			$this->log_dual("Received " . $userInfoString);
+
+
+			// Skip users without membership number
+			if (!is_numeric($userMembership) || !is_numeric($userContactId)) {
+				$this->log_dual("User is missing membership number or contact ID, skipping.");
+				continue;
+			}
 
 			//Create an array for the core wordpress user information
-			$user_data = array(
+			$accUserData = [
 				'first_name'	=>	$userFirstName,
 				'last_name'		=>	$userLastName,
-				'display_name'	=>	$userContactId,
+				'display_name'	=>	$userFirstName . " " . $userLastName,
 				'user_nicename'	=>	strtolower($userFirstName . "-" . $userLastName),
 				'user_login'	=>	$userContactId,
 				'user_email'	=>	$userEmail,
 				'user_pass'		=>	null,
-			);
+			];
+
+			$accUserMetaData = [
+				'home_phone' => $userHomePhone,
+				'cell_phone' => $userCellPhone,
+				'membership' => $userMembership,
+				'expiry' => $userExpiry,
+				'city' => $userCity
+			];
 			
-			//check if ID already exist
-			$user_id = username_exists($userContactId);
-			$update_this_user = false;
-			
-			if( is_numeric( $user_id ) ) {
-				//---------USER WAS FOUND IN DATABASE------------
-				$this->log_dual(" > found " . $userContactId . " (user #" . $user_id . ")");
-
-				//append ID to data object - wordpress will update the user instead of create a new one
-				$user_data = array_merge( array('ID' => $user_id), $user_data);
-
-				//Update DB only if something changed.
-				//Do a bunch of compare to decide if an update is needed.
-				//This logic needs to be extended each time we add a new user field.
-				// Get the user object
-				$user_meta = get_userdata($user_id);
-
-				// Get all the user roles as an array.
-				$user_roles = $user_meta->roles;
-				// Check if user has expired or ex-member role. If so, update role to default
-				if ( in_array( $role_expiry1, $user_roles, true ) ) {
-					$this->log_dual(" > User is " . $role_expiry1 . ", refreshing role to " . $default_role);
-					$update_this_user = true;
-					$user_data["role"] = $default_role;
-					$role_refreshed[] = $userContactId;
-					$role_refreshed_email[] = $userEmail;
-				} elseif (in_array( $role_expiry2, $user_roles, true ) ) {
-					$this->log_dual(" > User is " . $role_expiry2 . ", refreshing role to " . $default_role);
-					$update_this_user = true;
-					$user_data["role"] = $default_role;
-					$role_refreshed[] = $userContactId;
-					$role_refreshed_email[] = $userEmail;
-				}
+			// Check if ID or email already exist. Both should be unique 
+			$existingUser = get_user_by('login', $userContactId) ?: get_user_by('email', $userEmail);
```

Comment:
I think we unfortunately need to implement something a little more twisted, because of the scenario I sent by email where there are 4 family members with the same email address.  Here is the data to process:
243023	243025	France	Brunet	   claro046@gmail.com
243023	243024	Carl	Larocque   claro046@gmail.com
243023	243026	Aksel	Larocque   claro046@gmail.com
243023	243027	Elina	Larocque   claro046@gmail.com

I don't want each time the plugin runs to overwrite the same record with name France Brunet, then Carl Larocque, then Aksel Larocque then Elina Larocque.  This would look like the user changed names 4 times.  I think the new logic could to be something like this:

	$existingUser = get_user_by('login', $userContactId);
	if( !is_a( $existingUser, WP_User::class ) ) {
		//Not found by ContactID, search by email
		$existingUser = get_user_by('email', $userEmail);
		if( is_a( $existingUser, WP_User::class ) ) {
			//We found a user. If the name is the same, update it.
			//However if the name is different, it must be a family membership
			//sharing the same email, so abort, let the current user be.
 			if ($accUserData['display_name'] != $existingUser->display_name) continue;
		}
	}

What do you think? This would also allow a smooth upgrade of my database. On the first time the new plugin runs, it would not find anybody by ContactID, but would find by email and for those names that match, it would update the ContactID and any other fields that might have changed.